pkg/gatewayserver/io/ws/ws_test.go

```diff
@@ -1186,8 +1291,19 @@ func TestTraffic(t *testing.T) {
 						if err := json.Unmarshal(res, &msg); err != nil {
 							t.Fatalf("Failed to unmarshal response `%s`: %v", string(res), err)
 						}
-						msg.MuxTime = tc.ExpectedBSDownstream.(lbslns.DownlinkMessage).MuxTime
-						if !a.So(msg, should.Resemble, tc.ExpectedBSDownstream.(lbslns.DownlinkMessage)) {
+						expected := tc.ExpectedBSDownstream.(lbslns.DownlinkMessage)
+						msg.MuxTime = expected.MuxTime
+						if dlClass == ttnpb.Class_CLASS_A {
+							expected.XTime = clock.GetXTimeForTimestamp(timeStamp)
+						} else {
+							expected.XTime = clock.GetXTimeForTime(dlTime)
+							diff := expected.XTime - msg.XTime
+							if !a.So(diff, should.BeGreaterThanOrEqualTo, 0) && !a.So(diff, should.BeLessThan, 20000000) { //  Tolerance of 2s
```

Comment:
I ran some tests locally and this diff is anywhere between 0.2s to 1.2s. Since this whole PR is meant to essentially test this one value, I can't leave it out. But I can't guarantee that this won't be flaky with the CI.
I think the only way is to see how CI works and extend the tolerance if necessary.

Reply:
Where is the difference coming from at all ?

Reply:
I thought it was the TOA but that's pretty constant. I'll investigate this.

Reply:
So after some checks, the issue here seems to be that the mock gateway's concentrator timestamp and the concentrator timestamp of the GS are slightly out of sync each time and I'm unable to figure out why. I'll look at this fresh on Monday but the rest is good go to go.

Reply:
FTR: The flake was caused by the fact that `RxTime` does not use microsecond precision but the GS scheduler does.  So we now use `GatewayTime` only if there's GPSTime https://github.com/TheThingsNetwork/lorawan-stack/pull/5251/commits/c2106c5f20d88d391a5182985cf9b9d7ce434867.